### PR TITLE
5571: NativeThreadSet overhead in DataFileReader on highly concurrent reads

### PR DESCRIPTION
**Pull request by artemananiev**
Fix summary:

* `DataFileReader` (MerkleDb version only) is changed to use multiple file channels to avoid extra sync costs in FileChannelImpl.readInternal()
* Initially, a single file channel is open. Number of users (threads) is tracked, when it exceeds some threshold, a new file channel is open and used for further reads, up to a max number of file channels

Fixes: https://github.com/hashgraph/hedera-services/issues/5571
Signed-off-by: Artem Ananev <artem.ananev@swirldslabs.com>

### CODE REVIEW

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReader.java [hunk 1]
```diff
@@ -42,8 +44,28 @@
...
+    /** Max number of file channels to use for reading */
+    private static final int MAX_FILE_CHANNELS = 8;
+    /**
+     * When a data file reader is created, a single file channel is open to read data from the
+     * file. This channel is used by all threads. Number of threads currently reading data is
+     * tracked in {@link #fileChannelsInUse}. When the number of threads per opened file channel
+     * exceeds this threshold, a new file channel is open, unless there are {@link #MAX_FILE_CHANNELS}
+     * channels are already opened.
+     */
+    private static final int THREADS_PER_FILECHANNEL = 3;
```
Comment (imalygin) [inline platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReader.java:56]: Just curious, how did you come up with this value? Why is it 3 and not 5, let's say?

Comment (imalygin) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReader.java:56]: Also, would it be too wasteful to have a channel per thread?

Comment (imalygin) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReader.java:56]: @jsync-swirlds thank you for the clarification, it helps. The question about `THREADS_PER_FILECHANNEL` still stands, though.

Comment (artemananiev) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReader.java:56]: As discussed in slack, I'll increase it from 3 to 8. Changing to fully asynchronous mode is tracked in a separate epic, it's much more work (and design thoughts) than this simple PR.

**Review by imalygin — DISMISSED** [review]

**Review by OlegMazurov — APPROVED** [review]

**Review by OlegMazurov — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Pull request merged**